eo-maven-plugin/src/main/java/org/eolang/maven/optimization/ThrowingFunction.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016-2024 Objectionary.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven.optimization;
+
+/**
+ * Abstraction for XML optimizations.
+ * @param <I> Input type.
+ * @param <R> Return type.
+ * @param <E> Exception type.
+ * @since 0.35.0
+ */
+@FunctionalInterface
+public interface ThrowingFunction<I, R, E extends Exception> {
+    /**
+     * Applies this function to the given argument.
+     *
+     * @param input The function argument
+     * @return The function result
+     * @throws E If fails
+     */
+    R apply(I input) throws E;
```

Comment:
@Yanich96 I see two problems here:
_____ 
I believe that we don't need the `ThrowingFunction` interface at all because it's too generic and used only for optimization classes. Since we already have the `Optimization` abstraction, we don't need one more layer. It looks like overengineering. Please take a look at the [KISS principle](https://en.wikipedia.org/wiki/KISS_principle).
____

## (II) Changing `apply` method signature.
As I understood, the solution you provided was inspired by this [comment](https://github.com/objectionary/eo/issues/2713#issue-2053369477): 

> I think, we should change the interface of Optimization from Function<XML, XML> to Function<Path, XML>. It should accept the path of the source document, not the content of it. 

Well, I didn't write about it previously, so I will do it now. Changing the interface from `XML apply(XML)` to `XML apply(Path)` is dangerous for maintainability. At least, the solution overcomplicates the implementation of each separate optimization, as now each class has to "read" `XML` from the filesystem itself. I've counted four new usages of the `new XMLDocument(path)` instruction. Moreover, ideally, each `Optimization` has to save its results to a filesystem. Have you already counted how much it decreases the compilation time? I bet ~ `x2`.

Put it simply: 
We had 
```
XMIR -> (optimization) -> XMIR -> (optimization) -> XMIR
```

Now we have :
```
File -> XMIR -> (optimization) -> XMIR -> File -> XMIR -> (optimization) -> XMIR -> File
```
Looks overcomplicated, doesn't it?

Reply:
@volodya-lombrozo This task is fix OptCached and OptCachedTest. 

I deliberately left reading the name from the file so as not to break the program logic and tests. I created the issue in this PR in which I want to remove reading from a file. After removing reading from the file, the added interface and execution will not be needed.

Reply:
@Yanich96, can we implement this directly without adding any potentially dangerous code? It appears that we may not need this code at all. Why introduce this intermediate step?

Reply:
@volodya-lombrozo 
At the moment, I don't know how to avoid this code. But this code occurs when reading a XML file.
In my current task, only its name is taken from the XML file. The next step is that I want to get the program name from the absolute path, but I need to understand how the absolute paths of the program itself and its cache are obtained. In the next task the dangerous code will be removed.

Reply:
@Yanich96 Sorry, maybe I don't clearly understand you. Why do you need to change `XML apply(XML)` to `XML apply(Path)`?

Reply:
@volodya-lombrozo 
1) Only using the path can you find out the time of the last modification of the XML file.
2) Using the path allows us this allows us to not read the XML file. This will speed up the compiler.

Reply:
@Yanich96, maybe we can save the 'last modification time' to XMIR and simply utilize it?

As I've already mentioned, the solution significantly slows down the compiler due to unnecessary read/write operations from the file system. Consequently, I don't understand which speed up you mean.